modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/attachment/AttachmentUploadHelperImpl.java

```diff
@@ -0,0 +1,124 @@
+ * details.
+ */
+
+package com.liferay.object.internal.field.business.type.attachment;
+
+import com.liferay.document.library.kernel.model.DLFolderConstants;
+import com.liferay.object.field.business.type.attachment.AttachmentUploadHelper;
+import com.liferay.object.field.render.ObjectFieldRenderingContext;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Repository;
+import com.liferay.portal.kernel.portletfilerepository.PortletFileRepository;
+import com.liferay.portal.kernel.repository.model.Folder;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextFactory;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(immediate = true, service = AttachmentUploadHelper.class)
+public class AttachmentUploadHelperImpl implements AttachmentUploadHelper {
+
+	@Override
+	public Folder getFolder(
+			ObjectFieldRenderingContext objectFieldRenderingContext)
+		throws PortalException {
+
+		ServiceContext serviceContext = ServiceContextFactory.getInstance(
+			objectFieldRenderingContext.getHttpServletRequest());
+
+		serviceContext.setAddGroupPermissions(true);
+		serviceContext.setAddGuestPermissions(true);
+
+		Repository repository = _getRepository(
+			objectFieldRenderingContext.getGroupId(),
+			objectFieldRenderingContext.getPortletId(), serviceContext);
+
+		if (repository == null) {
+			return null;
+		}
+
+		return _getFolder(
+			objectFieldRenderingContext.getUserId(),
+			repository.getRepositoryId(), serviceContext);
+	}
+
+	private Folder _addFolder(
+		long userId, long repositoryId, ServiceContext serviceContext) {
+
+		try {
+			return _portletFileRepository.addPortletFolder(
+				userId, repositoryId,
+				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+				String.valueOf(userId), serviceContext);
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(portalException);
+			}
+
+			return null;
+		}
+	}
+
+	private Folder _getFolder(
+		long userId, long repositoryId, ServiceContext serviceContext) {
+
+		try {
+			return _portletFileRepository.getPortletFolder(
+				repositoryId, DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+				String.valueOf(userId));
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(portalException);
+			}
+
+			return _addFolder(userId, repositoryId, serviceContext);
+		}
+	}
+
+	private Repository _getRepository(
+		long groupId, String portletId, ServiceContext serviceContext) {
+
+		Repository repository = _portletFileRepository.fetchPortletRepository(
+			groupId, portletId);
+
+		if (repository == null) {
+			try {
+				return _portletFileRepository.addPortletRepository(
```

Comment:
I talked to Luiz about this and he told me that I could decide during the implementation, since the user will not see these folders when adding/editing an entry. I thought it was more "organized" this way.